### PR TITLE
.gitignore: ignore .elf

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Prevent firmware files from accidentally being commited.

### Issues/PRs references

#15247

### CODE REVIEW

**Comment by kaspar030** [comment]
https://github.com/RIOT-OS/RIOT/pull/12721

In this case, why would .hex, .bin, .elf be outside the bin/ folders?

**Comment by benpicco** [comment]
> In this case, why would .hex, .bin, .elf be outside the bin/ folders?

It happened at least twice within the last half year.
Adding them to `.gitignore` doesn't hurt and will prevent those little accidents in the future.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Object files
 *.o
+# firmware targets
+*.bin
```
Comment (aabadie) [inline .gitignore:4]: .bin is already covered by line 11

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+*.elf
+*.hex
```
Comment (aabadie) [inline .gitignore:6]: .hex are text files so easier to track in the change set of a PR, so I'm not sure it's needed.

**Comment by aabadie** [comment]
If the PR stays in this state, the commit message must be adapted (it's still referring to .bin and .hex files), same for the title of the PR.

**Review by aabadie — APPROVED** [review]
ACK!

**Pull request merged**